fastai/data_block.py

```diff
@@ -121,10 +121,11 @@ def from_df(cls, df:DataFrame, path:PathOrStr='.', cols:IntsOrStrs=0, processor:
         return res
 
     @classmethod
-    def from_csv(cls, path:PathOrStr, csv_name:str, cols:IntsOrStrs=0, header:str='infer', 
-                 processor:PreProcessors=None, **kwargs)->'ItemList':
-        "Create an `ItemList` in `path` from the inputs in the `cols` of `path/csv_name` opened with `header`."
-        df = pd.read_csv(Path(path)/csv_name, header=header)
+    def from_csv(cls, path:PathOrStr, csv_name:str, cols:IntsOrStrs=0, delimiter:str=None, header:str='infer', 
+                 col_names:Optional[Collection[str]]=None, processor:PreProcessors=None, **kwargs)->'ItemList':
```

Comment:
The `cols` argument can already either be a collection of strings that are the column names, so I don't want to add another one that would be redundant. You can check if `cols` is a list of strings with `is_listy(cols) and isinstance(cols[0],str)` for instance.

Reply:
The case where col_names is useful is where the csv has no header, so pandas doesn't know the column names. I don't think cols can overcome this (of course you can work around it by using only indices, which might be enough if there aren't many columns). Your call. 

I'll work on the other changes you mention.

Reply:
Oh right, cols would only contain the names of the inputs columns, not the outputs. If there are no col names, though, you have to know in advance what indexes your inputs are, so let's use integers in cols in this case.
I just don't want to have yet another argument if it's not necessary (`delimiter` is useful however).